### PR TITLE
WaterMask and [WIP]: Lighting

### PR DESCRIPTION
**Pull request by loicgasser**
Lighting URL:
1.0.0/ch.swisstopo.terrain.3d_light/default/20151231/4326/
Watermask URL:
1.0.0/ch.swisstopo.terrain.3d_water/default/20151231/4326/

We're using a small extent for zooms 8 to 17.

### CODE REVIEW

File: forge/terrain/topology.py [hunk 1]
```diff
@@ -53,13 +56,20 @@ def addVertices(self, vertices):
             lookupKey = ','.join([str(vertex[0]), str(vertex[1]), str(vertex[2])])
             faceIndex = self._lookupVertexIndex(lookupKey)
             if faceIndex is not None:
+                # Sometimes we can have triangles with zero area (due to unfortunate clipping)
+                # In that case skip them
+                if faceIndex in face:
+                    break
```
Comment (loicgasser) [inline forge/terrain/topology.py]: Fishy no? I recreated the tiles for 8 to 14 as I think we now have border atefacts with this approach.

File: forge/lib/oct_encoding.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    res = [x, y, 0.0]
+    res[0] = fromSnorm(x)
+    res[1] = fromSnorm(y)
+    res[2] = 1.0 - (abs(res[1]) - abs(res[1]))
+
+    if res[2] < 0.0:
+        oldX = res[0]
+        res[0] = (1.0 - abs(res[1]) * signNotZero(oldX))
+        res[1] = (1.0 - abs(oldX) * signNotZero(res[1]))
+    return c3d.normalize(res)
```
Comment (gjn) [inline forge/lib/oct_encoding.py]: Is this encoding/decoding based on https://github.com/AnalyticalGraphicsInc/cesium/blob/b161b6429b9201c99e5fb6f6e6283f3e8328b323/Source/Core/AttributeCompression.js#L43 ? Or did you use a different source?

Comment (loicgasser) [inline forge/lib/oct_encoding.py]: yes it is.

**Comment by loicgasser** [comment]
This is basically the new master I'll go along and merge and fix light afterwards.

**Pull request merged**